src/types/package.rs

```diff
@@ -450,6 +482,107 @@ pub struct PackageDescriptor {
     pub package_type: PackageType,
 }
 
+/// Describes a package descriptor in the system with lockfile path
+#[derive(
+    PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug, Serialize, Deserialize, JsonSchema,
+)]
+pub struct LockFilePackageDescriptor {
+    pub package_descriptor: PackageDescriptor,
+    pub lockfile_path: Option<String>,
+}
+
+impl From<&PackageDescriptor> for LockFilePackageDescriptor {
+    fn from(value: &PackageDescriptor) -> Self {
+        LockFilePackageDescriptor {
+            package_descriptor: value.clone(),
+            lockfile_path: None,
+        }
+    }
+}
+
+/// Describes a package specifier in the system with lockfile path
+#[derive(
+    PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug, Serialize, Deserialize, JsonSchema,
+)]
+pub struct LockFilePackageSpecifier {
+    pub package_specifier: PackageSpecifier,
+    pub lockfile_path: Option<String>,
+}
+
+impl From<&PackageSpecifier> for LockFilePackageSpecifier {
+    fn from(value: &PackageSpecifier) -> Self {
+        LockFilePackageSpecifier {
+            package_specifier: value.clone(),
+            lockfile_path: None,
+        }
+    }
+}
+
+/// Describes a package url in the system with lockfile path
+#[derive(
+    PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug, Serialize, Deserialize, JsonSchema,
+)]
+pub struct ParsedPurl {
+    pub purl: String,
+    pub lockfile_path: Option<String>,
+}
+
+impl TryFrom<&PackageDescriptor> for ParsedPurl {
+    type Error = purl::PackageError;
+
+    fn try_from(value: &PackageDescriptor) -> Result<Self, Self::Error> {
+        let package_type = purl::PackageType::from(value.package_type);
+        let purl = Purl::builder(package_type, &value.name)
+            .with_version(&value.version)
+            .build()?
+            .to_string();
+        Ok(ParsedPurl {
+            purl,
+            lockfile_path: None,
+        })
+    }
+}
+
+impl TryFrom<&ParsedPurl> for PackageDescriptor {
+    type Error = purl::PackageError;
+
+    fn try_from(value: &ParsedPurl) -> Result<Self, Self::Error> {
+        let purl = Purl::from_str(&value.purl)?;
+        let package_type = PackageType::try_from(purl.package_type())?;
+
+        let package_name = match purl.namespace() {
+            Some(namespace) => {
+                let name = purl.name();
+                if package_type == PackageType::Maven {
+                    let ns = namespace.replace('/', ".");
+                    format!("{ns}:{name}")
+                } else {
+                    format!("{namespace}/{name}")
+                }
+            }
+            None => purl.name().into(),
+        };
+
+        let vcs_qualifier = purl.qualifiers().get("vcs_url");
```

Comment:
I don't think we should be trying to convert between `vcs_url` and version like this.

Reply:
This is another [Phylum-ism](https://github.com/phylum-dev/cli/blob/e232419d80195178eddd738fef18dd2751212e81/cli/src/commands/parse.rs#L206)

Reply:
I think in the case of dependencies where the version is specified in the form of a VCS revision in the ecosystem's native format, it's probably better to use that as the version than to put it into a `vcs_url` qualifier. `vcs_url` seems more useful for cases where the ecosystem doesn't support that or where you're specifying both a version and a `vcs_url` at the same time.

But I guess for now it doesn't make much of a difference because we don't analyze packages pulled from arbitrary VCS URLs. Whether the version is in the version as the backend expects or only in the `vcs_url`, the package won't process. However, we do need to make sure whatever PURLs are submitted do have a version that is recognized by the backend because the version is a mandatory field on the dependency.

Reply:
Actually, this isn't needed after all. Removing it.